packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts

```diff
@@ -349,6 +349,16 @@ export default createRule<Options, MessageIds>({
 
           const contextualType = getContextualType(checker, originalNode);
           if (contextualType) {
+            const typeIsUnknown = isTypeFlagSet(type, ts.TypeFlags.Unknown);
+            const contextualTypeIsUnknown = isTypeFlagSet(
+              contextualType,
+              ts.TypeFlags.Unknown,
+            );
+
+            if (typeIsUnknown && !contextualTypeIsUnknown) {
+              return;
+            }
```

Comment:
[Refactor] If `typeIsUnknown` then we don't need to check `!contextualTypeIsUnknown`, right? I'm sure this never actually matters for perf, but just making the code IMO a bit more representative of what it's trying to do:

```suggestion
            if (
              isTypeFlagSet(type, ts.TypeFlags.Unknown) &&
              !isTypeFlagSet(contextualType, ts.TypeFlags.Unknown)
            ) {
              return;
            }
```